### PR TITLE
Replace assertEquals by assertSame

### LINKED ISSUE #430: Replace assertEquals with assertSame where possible
I just learned that `assertEquals` of PHPUnit only uses the weak comparison `==`. So something like this would not fail:
```php
$this->assertEquals(['depth' => 0], ['depth' => null]);
```

`assertSame` uses the strong check `===` and we should use it as much as possible. Replace all `assertEquals` with `assertSame` in the tests. Then investigate the failing tests if `assertEquals` is actually required there (or if this is a legitimate error and something must be fixed).

### PR DESCRIPTION
**Pull request by lehecht**
Use strong checks for better testing.

Closes #430.

### CODE REVIEW

**Comment by lehecht** [comment]
@mzur 
I overlooked a few files. I'll message you once I'm done here.

**Comment by lehecht** [comment]
@mzur 
The code is ready to be reviewed.

**Comment by mzur** [comment]
I'll have a look. Please work on the open [tasks](https://github.com/orgs/biigle/projects/4/views/9) first before starting something else 😉

File: tests/php/Http/Controllers/Api/ProjectInvitationControllerTest.php [hunk 1]
```diff
@@ -38,15 +38,15 @@ public function testStore()
                 'expires_at' => $timestamp,
             ])
             ->assertSuccessful();
-
+    
         $invitation = $this->project()->invitations()->first();
         $this->assertNotNull($invitation);
-        $this->assertEquals($timestamp, $invitation->expires_at);
-        $this->assertEquals(0, $invitation->current_uses);
+        $this->assertSame($timestamp->toDateTimeString(), $invitation->expires_at->tz($timestamp->timezone)->toDateTimeString());
```
Comment (mzur) [inline tests/php/Http/Controllers/Api/ProjectInvitationControllerTest.php:44]: Just keep `assertEquals` for simplicity here.

File: tests/php/Http/Controllers/Api/Volumes/MetadataControllerTest.php [hunk 2]
```diff
@@ -60,15 +60,15 @@ public function testStoreImageMetadata()
...
+        $this->assertSame(28.123, $jpg->lat);
+        $this->assertSame(-1500, intval($jpg->metadata['gps_altitude']));
+        $this->assertSame(2.6, floatval($jpg->metadata['area']));
         // Import should update but not destroy existing metadata.
-        $this->assertEquals(10, $jpg->metadata['distance_to_ground']);
-        $this->assertEquals(4000, $jpg->metadata['water_depth']);
-        $this->assertEquals(180, $jpg->metadata['yaw']);
+        $this->assertSame(10, intval($jpg->metadata['distance_to_ground']));
+        $this->assertSame(4000, $jpg->metadata['water_depth']);
+        $this->assertSame(180, intval($jpg->metadata['yaw']));
```
Comment (mzur) [inline tests/php/Http/Controllers/Api/Volumes/MetadataControllerTest.php:71]: Please update the metadata code so these values are actually stored as int/float.

File: tests/php/Jobs/CreateNewImagesOrVideosTest.php [hunk 3]
```diff
@@ -74,13 +74,13 @@ public function testHandleImageMetadata()
...
-        $this->assertEquals(2.6, $image->metadata['area']);
-        $this->assertEquals(10, $image->metadata['distance_to_ground']);
-        $this->assertEquals(180, $image->metadata['yaw']);
+        $this->assertSame('2016-12-19 12:27:00', $image->taken_at->toDateTimeString());
+        $this->assertSame(52.220, $image->lng);
+        $this->assertSame(28.123, $image->lat);
+        $this->assertSame(-1500, intval($image->metadata['gps_altitude']));
+        $this->assertSame(2.6, floatval($image->metadata['area']));
+        $this->assertSame(10, intval($image->metadata['distance_to_ground']));
+        $this->assertSame(180, intval($image->metadata['yaw']));
```
Comment (mzur) [inline tests/php/Jobs/CreateNewImagesOrVideosTest.php:83]: Same as above.

File: tests/php/Jobs/ProcessNewImageTest.php [hunk 4]
```diff
@@ -76,8 +76,8 @@ public function testHandleCollectMetadataAreaYaw()
         $job->handle();
         $image = $image->fresh();
 
-        $this->assertEquals(47.75, $image->metadata['yaw']);
-        $this->assertEquals(2.6, $image->metadata['area']);
+        $this->assertSame(47.75, floatval($image->metadata['yaw']));
+        $this->assertSame(2.6, floatval($image->metadata['area']));
```
Comment (mzur) [inline tests/php/Jobs/ProcessNewImageTest.php:80]: Same as above.

File: tests/php/Jobs/ProcessNewVideoTest.php [hunk 5]
```diff
@@ -21,8 +21,8 @@ public function testHandleThumbnails()
         $job->duration = 10;
 
         $job->handle();
-        $this->assertEquals(10, $video->fresh()->duration);
-        $this->assertEquals([0.5, 5, 9.5], $job->times);
+        $this->assertSame(10, intval($video->fresh()->duration));
```
Comment (mzur) [inline tests/php/Jobs/ProcessNewVideoTest.php:24]: Please update the code so the duration is always returned as float.

File: tests/php/Services/Auth/ApiGuardTest.php [hunk 6]
```diff
@@ -120,7 +120,7 @@ public function testTouchToken()
         $token->updated_at = Carbon::now(-5);
         $token->save();
 
-        $this->assertEquals($token->updated_at, $token->fresh()->updated_at);
+        $this->assertSame($token->updated_at->toDateTimeString(), $token->fresh()->updated_at->toDateTimeString());
```
Comment (mzur) [inline tests/php/Services/Auth/ApiGuardTest.php:123]: Keep using `assertEquals` for simplicity.

**Review by mzur — CHANGES_REQUESTED** [review]

**Review by mzur — APPROVED** [review]

**Pull request merged**